nvs: Add forgotten NVS cache rebuild

jdascenzio opened the pull request:
During the NVS initialization, if gc had to be done, the NVS cache rebuild wasn't called.

Signed-off-by: Julien D'Ascenzio <julien.dascenzio@paratronic.fr>

Laczen commented:
@jdascenzio, while doing gc the cache should be rebuild automatically. Have you found a situation where the cache was not updated?

jdascenzio commented:
> @jdascenzio, while doing gc the cache should be rebuild automatically. Have you found a situation where the cache was not updated?

For me, the cache isn't rebuild in gc. I was able to reproduce this situation by forcing my application to stop during gc. And then, at the next boot, I could see that nvs cache wasn't initialized at the exit of nvs_init

jdascenzio committed:
nvs: NVS cache always rebuild on successful NVS initialization

During the NVS initialization, if gc had to be done, the NVS cache
rebuild wasn't called.

Signed-off-by: Julien D'Ascenzio <julien.dascenzio@paratronic.fr>

jdascenzio commented:
> > For me, the cache isn't rebuild in gc. I was able to reproduce this situation by forcing my application to stop during gc. And then, at the next boot, I could see that nvs cache wasn't initialized at the exit of nvs_init
> 
> OK, you are right. I think it would be better to move lines 890-892 to line 903 and to use something like:
> 
> ```
>         if ((!rc) && IS_ENABLED(CONFIG_NVS_LOOKUP_CACHE) {
>                 rc = nvs_lookup_cache_rebuild(fs);
>         }
>         k_mutex_unlock(&fs->nvs_lock);
> ```

I rewrote the code based on your suggestions.
